gammapy/modeling/sampler.py

```diff
@@ -0,0 +1,226 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Sampler parameter classes."""
+
+from .utils import _parse_datasets
+
+__all__ = ["Sampler", "SamplerLikelihood", "SamplerResult"]
+
+
+class Sampler:
+    """Sampler class.
+
+    The sampler class provides a uniform interface to multiple sampler backends.
+    Currently available: "UltraNest", ("zeusmc", "emcee"  in #TODO).
+
+    Parameters
+    ----------
+    backend : {"ultranest"}
+        Global backend used for sampler. Default is "ultranest".
+        UltraNest: Most options can be found in the UltraNest doc
+
+    sampler_opts : dict, optional
+        Sampler options passed to the sampler.
+        Noteworthy options:
```

Comment:
This is more than that. They are options for which we (gammapy) fix default values. One should also precise the used default values.

Reply:
indeed teh defaults should be explicited 
Is there a gammapy common syntax for defaults params  ?
live_points : int  (default : 0.5) 
or something better ?

Reply:
Done following what was in iMinuit or fit docstring